### PR TITLE
📖 Update the documentation for v1alpha3

### PR DESCRIPTION
**Pull request by maelk**
**What this PR does / why we need it**:
This PR updates the documentation for v1alpha3 and adds the releasing guidelines as started by @Arvinderpal in https://github.com/metal3-io/cluster-api-provider-baremetal/pull/226 (inactive)

### CODE REVIEW

File: docs/deployment_workflow.md [hunk 1]
```diff
@@ -34,24 +36,24 @@ following the correct flow.
...
+   BareMetalMachine and KubeadmConfig.
+1. CABPK will generate the cloud-init output for this machine and create a
+   secret containing it.
+1. The CAPI controller will copy the userData secret name into the machine
+   object and set the bootstrapReady field to true.
+1. Once the machine has userdataSecretName, OwnerRef and bootstrapReady properly
+   set, the CAPBM controller will select, if possible, a BareMetalHost that
+   matches the criteria, or wait until one is available. If matched, the CAPBM
+   controller will create a secret with the userData. and set the BareMetalHost
+   spec accordingly to the BareMetalMachine specs.
```
Comment (Xenwar) [inline docs/deployment_workflow.md]: Should these numbers increase ?
1. ...
2..

File: docs/releasing.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+1. Create the release notes `make release-notes`. Copy the output and sort
+   manually the items that need to be sorted.
+1. Create an annotated tag `git tag -a v0.x.y -m v0.x.y`. To use your GPG
+   signature when pushing the tag, use `git tag -s [...]` instead
+1. Push the tag to the GitHub repository `git push origin v0.x.y`
+   NB: `origin` should be the name of the remote pointing to
+   `github.com/metal3-io/cluster-api-provider-baremetal`
+1. Run `make release` to build artifacts (the image is automatically built by CI)
+1. Run `make release` to build artifacts
+1. [Create a release in GitHub](https://help.github.com/en/github/administering-a-repository/creating-releases)
```
Comment (Xenwar) [inline docs/releasing.md]: same here

File: docs/deployment_workflow.md (hunk 1 above)
Comment (maelk) [reply docs/deployment_workflow.md]: No, this is done automatically when rendered

File: docs/releasing.md (hunk 2 above)
Comment (maelk) [reply docs/releasing.md]: same answer :D

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+1. Create the release notes `make release-notes`. Copy the output and sort
+   manually the items that need to be sorted.
+1. Create an annotated tag `git tag -a v0.x.y -m v0.x.y`. To use your GPG
+   signature when pushing the tag, use `git tag -s [...]` instead
+1. Push the tag to the GitHub repository `git push origin v0.x.y`
+   NB: `origin` should be the name of the remote pointing to
+   `github.com/metal3-io/cluster-api-provider-baremetal`
+1. Run `make release` to build artifacts (the image is automatically built by CI)
+1. Run `make release` to build artifacts
```
Comment (kashifest) [inline docs/releasing.md]: Duplicate ?

File: README.md [hunk 4]
```diff
@@ -34,37 +34,32 @@ for testing purposes only, when Baremetal Operator is not deployed
     make deploy-bmo-cr
 ```
 
-### Deploy CAPM3 CRDs
+### Deploy CAPM3
 
-Deploys CAPM3 CRDs
+Deploys CAPBM CRDs and deploys CAPI, CABPK, CAKCP and CAPBM controllers
```
Comment (Xenwar) [inline README.md]: should CAPBM be changed by CAPM3 ?

File: docs/deployment_workflow.md [hunk 5]
```diff
@@ -34,24 +36,24 @@ following the correct flow.
...
-   accordingly to the BareMetalMachine specs.
+1. The CAPI controller will set the OwnerRef
+1. The KubeadmControlPlane controller will wait until the cluster has
+   infrastructureReady set to true, and generate the first machine,
+   BareMetalMachine and KubeadmConfig.
+1. CABPK will generate the cloud-init output for this machine and create a
+   secret containing it.
+1. The CAPI controller will copy the userData secret name into the machine
+   object and set the bootstrapReady field to true.
+1. Once the machine has userdataSecretName, OwnerRef and bootstrapReady properly
```
Comment (Xenwar) [inline docs/deployment_workflow.md]: property ?
if properly ,what are the proper values ?

File: docs/deployment_workflow.md [hunk 6]
```diff
@@ -64,10 +66,13 @@ An outline of the workflow is below.
    the providerID of the Machine, copied from the BaremetalMachine. If matching,
    the control plane initialized status will be set to true and the machine
    state to running.
-1. CAPBK will then generate the userData for all the other nodes and a similar
-   workflow will happen for their deployment.
+1. CAKCP will then do the same for each further controller node until reaching
+   the desired replicas number, one by one, triggering the same workflow.
```
Comment (Xenwar) [inline docs/deployment_workflow.md]: replica number

**Pull request merged**